internal/controller/fluxinstance_controller.go

```diff
@@ -223,6 +223,17 @@ func (r *FluxInstanceReconciler) build(ctx context.Context,
 		}
 	}
 
+	if obj.Spec.Sync != nil {
```

Comment:
reflecting a checksum in `status` could be useful here.

Reply:
This is already the case, the digest included in status captures any change to spec.